plugins/woocommerce/tests/php/includes/class-wc-tracker-test.php

```diff
@@ -305,4 +305,54 @@ public function test_get_tracking_data_admin_install_timestamp() {
 	public function test_get_tracking_data_snapshot_generation_time() {
 		$this->assertGreaterThan( 0, WC_Tracker::get_tracking_data()['snapshot_generation_time'] );
 	}
+
+	/**
+	 * @testDox Test woocommerce_allow_tracking related data is included in tracking snapshot.
+	 */
+	public function test_tracking_data_woocommerce_allow_tracking() {
+		$current_woocommerce_allow_tracking = get_option( 'woocommerce_allow_tracking', 'no' );
+
+		// Clear everything.
+		update_option( 'woocommerce_allow_tracking', 'no' );
+		delete_option( 'woocommerce_allow_tracking_last_modified' );
+		delete_option( 'woocommerce_allow_tracking_first_optin' );
+
+		$tracking_data = WC_Tracker::get_tracking_data();
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking', $tracking_data );
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking_last_modified', $tracking_data );
+		$this->assertArrayHasKey( 'woocommerce_allow_tracking_first_optin', $tracking_data );
+
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'no' );
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking_last_modified'], 'unknown' );
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking_first_optin'], 'unknown' );
+
+		$time_one = time();
+		update_option( 'woocommerce_allow_tracking', 'yes' );
+		$tracking_data = WC_Tracker::get_tracking_data();
+		$this->assertEquals( $tracking_data['woocommerce_allow_tracking'], 'yes' );
+		$this->assertTrue( $tracking_data['woocommerce_allow_tracking_last_modified'] >= $time_one );
```

Comment:
`>=` is a very broad range to check against. I understand we don't want to introduce any flakiness, if it jumps to the next second right in the middle of this test. But should we add an additional check for `< $time_one + MINUTE_IN_SECONDS`?